data/src/test/java/org/apache/iceberg/data/TestPartitionStatsWriterUtil.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.data;
+
+import static org.apache.iceberg.PartitionStatsUtil.Column;
+
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Random;
+import org.apache.iceberg.Files;
+import org.apache.iceberg.PartitionData;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.PartitionStatsUtil;
+import org.apache.iceberg.Partitioning;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SortOrder;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TestTables;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.types.Types;
+import org.assertj.core.api.Assertions;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class TestPartitionStatsWriterUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(TestPartitionStatsWriterUtil.class);
+
+  private static final Schema SCHEMA =
+      new Schema(
+          Types.NestedField.required(1, "id", Types.LongType.get()),
+          Types.NestedField.optional(2, "data", Types.StringType.get()),
+          Types.NestedField.optional(3, "binary", Types.BinaryType.get()));
+
+  @Rule public TemporaryFolder temp = new TemporaryFolder();
+
+  @Test
+  public void testPartitionStats() throws Exception {
+    PartitionSpec spec = PartitionSpec.builderFor(SCHEMA).identity("id").build();
+    Table testTable =
+        TestTables.create(
+            temp.newFolder("test_partition_stats"),
+            "test_partition_stats",
+            SCHEMA,
+            spec,
+            SortOrder.unsorted(),
+            2);
+
+    Types.StructType partitionSchema = Partitioning.partitionType(testTable);
+    Schema dataSchema = PartitionStatsUtil.schema(partitionSchema);
+
+    ImmutableList.Builder<Record> partitionListBuilder = ImmutableList.builder();
+
+    long seed = System.currentTimeMillis();
```

Comment:
In general, unless we are doing an explicit property test we tend to make sure the tests are consistent. So I would just freeze the seed as a constant in the file.